pkg/trace/obfuscate/sql.go

```diff
@@ -358,3 +358,25 @@ func (o *Obfuscator) obfuscateSQL(span *pb.Span) {
 	}
 	traceutil.SetMeta(span, sqlQueryTag, oq.Query)
 }
+
+// ObfuscateSQLStringSafe obfuscates the sql string, returning either the successfully obfuscated string or a string
+// containing an error message explaining why the obfuscation failed
+func (o *Obfuscator) ObfuscateSQLStringSafe(sqlString string) string {
```

Comment:
There are way too many of these. We have:
* `obfuscateSQL`
* `obfuscateSQLString`
* `ObfuscateSQLString`

I think that's enough :) 

Instead, let's add a method onto `jsonObfuscator` called perhaps `obfuscateSQL`:

```go
// sqlObfuscator returns an SQL obfuscation function which uses o as its Obfuscator. It returns
// either the successfully obfuscated string or a string containing an error message explaining
// why the obfuscation failed.
func (*jsonObfuscator) sqlObfuscator(o *Obfuscator) func(string) string {
    return func(in string) string {
        // ... contents of this function
    }
}
```
And create this as the transformer.

What do you think?

Reply:
I agree that there are many of them but I'm not sure that this approach would make things clearer. If anything it may get more convoluted because it's a function called `sqlObfuscator` that takes an `obufscator` and presumably returns an "obfuscator" that's not actually an obfuscator in the way the word is used in the rest of this module, rather it's a function that obfuscates only strings. 🤷‍♂️

Reply:
I understand, but this is way too much, and it makes no sense to have one more. This method belongs on `*jsonObfuscator`. If you can find an alternative way that's fine but I feel pretty strongly about moving it away from here. I don't find it that confusing. Call it `tranformerFunc` if you will :)

Reply:
Ok, I see what you mean. Refactored here https://github.com/DataDog/datadog-agent/pull/5692/commits/d7064f556056480bfe5d020295f2e5a9dea2f2e9 does that work?